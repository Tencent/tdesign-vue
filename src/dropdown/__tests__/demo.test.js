/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/src/dropdown/_example/base.vue';
import buttonVue from '@/src/dropdown/_example/button.vue';
import customVue from '@/src/dropdown/_example/custom.vue';
import disabledVue from '@/src/dropdown/_example/disabled.vue';
import eventVue from '@/src/dropdown/_example/event.vue';
import longVue from '@/src/dropdown/_example/long.vue';
import multipleVue from '@/src/dropdown/_example/multiple.vue';
import slotVue from '@/src/dropdown/_example/slot.vue';
import splitVue from '@/src/dropdown/_example/split.vue';

const mapper = {
  baseVue,
  buttonVue,
  customVue,
  disabledVue,
  eventVue,
  longVue,
  multipleVue,
  slotVue,
  splitVue,
};

describe('Dropdown', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Dropdown ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
