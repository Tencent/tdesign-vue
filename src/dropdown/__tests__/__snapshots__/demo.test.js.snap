// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Dropdown baseVue demo works fine 1`] = `
<div
  class="tdesign-demo-dropdown"
>
  <button
    class="t-button t-size-m t-button--variant-text t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      <span
        class="tdesign-demo-dropdown__text"
      >
        
        更多
        
        <svg
          class="t-icon t-icon-chevron-down"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M3.54 6.46l.92-.92L8 9.08l3.54-3.54.92.92L8 10.92 3.54 6.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;

exports[`Dropdown Dropdown buttonVue demo works fine 1`] = `
<div>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--shape-square"
    type="button"
  >
    <span
      class="t-button__text"
    >
      <svg
        class="t-icon t-icon-ellipsis"
      >
        <use
          href="#t-icon-ellipsis"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Dropdown Dropdown customVue demo works fine 1`] = `
<div>
  <button
    class="t-button t-size-m t-button--variant-text t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      下拉菜单
    </span>
  </button>
</div>
`;

exports[`Dropdown Dropdown disabledVue demo works fine 1`] = `
<div
  class="tdesign-demo-dropdown"
>
  <button
    class="t-button t-size-m t-button--variant-text t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      <span
        class="tdesign-demo-dropdown__text"
      >
        
        下拉菜单
        
        <svg
          class="t-icon t-icon-chevron-down"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M3.54 6.46l.92-.92L8 9.08l3.54-3.54.92.92L8 10.92 3.54 6.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;

exports[`Dropdown Dropdown eventVue demo works fine 1`] = `
<div>
  <button
    class="t-button t-size-m t-button--variant-text t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      下拉菜单
    </span>
  </button>
</div>
`;

exports[`Dropdown Dropdown longVue demo works fine 1`] = `
<div>
  <button
    class="t-button t-size-m t-button--variant-text t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      下拉菜单
    </span>
  </button>
</div>
`;

exports[`Dropdown Dropdown multipleVue demo works fine 1`] = `
<div
  class="tdesign-demo-dropdown"
>
  <button
    class="t-button t-size-m t-button--variant-text t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      <span
        class="tdesign-demo-dropdown__text"
      >
        
        下拉菜单
        
        <svg
          class="t-icon t-icon-chevron-down"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M3.54 6.46l.92-.92L8 9.08l3.54-3.54.92.92L8 10.92 3.54 6.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;

exports[`Dropdown Dropdown slotVue demo works fine 1`] = `
<div
  class="t-space t-space-vertical"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <div
      class="t-space t-space-vertical"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <h4>
          支持直接使用 t-dropdown-menu
        </h4>
      </div>
      <div
        class="t-space-item"
      >
        <span>
          <button
            class="t-button t-size-m t-button--variant-text t-button--theme-default"
            type="button"
          >
            <span
              class="t-button__text"
            >
              下拉菜单
            </span>
          </button>
           
          <div
            class="t-dropdown__menu t-dropdown__menu--right"
            style="max-height: 300px;"
          />
        </span>
      </div>
    </div>
  </div>
  <div
    class="t-space-item"
  >
    <div
      class="t-space t-space-vertical"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <h4>
          兼容历史版本 通过 dropdown 具名插槽的使用
        </h4>
      </div>
      <div
        class="t-space-item"
      >
        <button
          class="t-button t-size-m t-button--variant-text t-button--theme-default"
          type="button"
        >
          <span
            class="t-button__text"
          >
            下拉菜单
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dropdown Dropdown splitVue demo works fine 1`] = `
<div>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      下拉菜单
    </span>
  </button>
</div>
`;
